src/farcasterd/runtime.rs

```diff
@@ -97,84 +83,52 @@ pub fn run(
-    public_offers: HashSet<PublicOffer>,
-    arb_addrs: HashMap<Uuid, bitcoin::Address>,
-    acc_addrs: HashMap<Uuid, monero::Address>,
-    consumed_offers: HashMap<PublicOffer, (SwapId, ServiceId)>,
-    node_ids: HashMap<Uuid, NodeId>, // Only populated by maker. TODO is it possible? HashMap<SwapId, PublicKey>
-    peerd_ids: HashMap<Uuid, ServiceId>, // Only populated by maker.
-    wallet_token: Token,
-    pending_sweep_requests: HashMap<ServiceId, Request>, // TODO: merge this with pending requests eventually
-    syncers: Syncers,
-    progress: HashMap<ServiceId, VecDeque<Request>>,
-    progress_subscriptions: HashMap<ServiceId, HashSet<ServiceId>>,
-    funding_btc: HashMap<SwapId, (bitcoin::Address, bitcoin::Amount, bool)>,
-    funding_xmr: HashMap<SwapId, (monero::Address, monero::Amount, bool)>,
-    checkpointed_pub_offers: List<CheckpointEntry>,
-    restoring_swap_id: HashSet<SwapId>,
-    stats: Stats,
-    config: Config,
-    syncer_task_counter: u32,
-    syncer_tasks: HashMap<TaskId, ServiceId>,
+    pub identity: ServiceId,                     // Set on Runtime instantiation
+    pub wallet_token: Token,                     // Set on Runtime instantiation
+    pub started: SystemTime,                     // Set on Runtime instantiation
+    pub node_secret_key: Option<SecretKey>, // Set by Keys request shortly after Hello from walletd
+    pub node_public_key: Option<PublicKey>, // Set by Keys request shortly after Hello from walletd
+    pub listens: HashSet<InetSocketAddr>,   // The socket address of the binding peerd listeners
+    pub spawning_services: HashSet<ServiceId>, // Services that have been launched, but have not replied with Hello yet
+    pub registered_services: HashSet<ServiceId>, // Services that have announced themselves with Hello
+    pub public_offers: HashSet<PublicOffer>,     // The set of all known public offers
+    pub progress: HashMap<ServiceId, VecDeque<Request>>, // A mapping from Swap ServiceId to its sent and received progress requests
+    pub progress_subscriptions: HashMap<ServiceId, HashSet<ServiceId>>, // A mapping from a Client ServiceId to its subsribed swap progresses
+    pub checkpointed_pub_offers: List<CheckpointEntry>, // A list of existing swap checkpoint entries that may be restored again
+    pub stats: Stats,                                   // Some stats about offers and swaps
+    pub config: Config, // Configuration for syncers, auto-funding, and grpc
+    pub syncer_task_counter: u32, // A strictly incrementing counter of issued syncer tasks
+    pub trade_state_machines: Vec<TradeStateMachine>, // New trade state machines are inserted on creation and destroyed upon state machine end transitions
+    pub syncer_state_machines: HashMap<TaskId, SyncerStateMachine>, // New syncer state machines are inserted by their syncer task id when sending a syncer request and destroyed upon matching syncer request receival
 }
 
 impl CtlServer for Runtime {}
 
 #[derive(Default)]
```

Comment:
ahh! so much :cut_of_meat:  trimmed :100:

once you're done with other changes, reminder to reduce visibility of the fields here, would be nice to have at most `pub(super)` since afaict these are only accessed by `farcasterd::{syncer, trade}_state_machine` (ditto for other funcs in `runtime`)

Reply:
I prefer not add these qualifiers as part of this PR. We don't use them anywhere else in the code where we similarly export structs and member functions, and their benefit is questionable, since we are not writing a library here. I did make sure though that only those functions and fields are pub that need to be in 04997366cc2419fc04fbbd1f8059079eb9d03c10 and afb0647cf9a8ff4c1eba3052fb735d6f54f1973d .

Reply:
Thanks for cleaning up.

The `super` qualifiers have more purpose in a library since that protects misuse by external projects - the goal here would be to protect development within the node.
But thinking about it now: since the node's segregated into microservices, the natural visibility scope for most `pub` things (and in particular the ones here) is effectively already `pub(super)`.